src/mongo_client.ts

```diff
@@ -477,6 +477,11 @@ export class MongoClient extends TypedEventEmitter<MongoClientEvents> implements
     return this.s.bsonOptions;
   }
 
+  /** @internal */
+  get timeoutMS(): number | undefined {
+    return this.options.timeoutMS;
+  }
```

Comment:
Added to ensure that `resolveOptions` options inheritance is actually able to get option from `MongoClient`